lib/OpenQA/WebAPI/Controller/API/V1/Worker.pm

```diff
@@ -42,14 +43,18 @@ websocket status.
 
 sub list {
     my ($self) = @_;
+    my $limits = OpenQA::App->singleton->config->{misc_limits};
+    my $limit = min($limits->{generic_max_limit}, $self->param('limit') // $limits->{generic_default_limit});
     my $validation = $self->validation;
     $validation->optional('live')->num(1);
     return $self->reply->validation_error({format => 'json'}) if $validation->has_error;
     my $live = $validation->param('live');
     my $workers = $self->schema->resultset("Workers");
     my $ret = [];
+    my $i = 0;
 
-    while (my $w = $workers->next) {
+    while (my $w = $workers->next and $i < $limit) {
+        $i++;
```

Comment:
Yes, that would make sense. (The code is filtering jobs with falsy ID out in Perl level but it shouldn't bother us here because that is most likely useless anyways and could actually be removed.)